staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/status_strategy_test.go

```diff
@@ -136,3 +139,182 @@ func TestPrepareForUpdate(t *testing.T) {
 		}
 	}
 }
+
+func strPtr(s string) *string { return &s }
+
+func TestStatusStrategyValidateUpdate(t *testing.T) {
+	strategy := statusStrategy{}
+	kind := schema.GroupVersionKind{
+		Version: "v1",
+		Kind:    "foo",
+		Group:   "test",
+	}
+	strategy.customResourceStrategy.validator.kind = kind
+	v := "v1"
+	structuralSchemas := map[string]*structuralschema.Structural{}
+	structuralSchema, _ := structuralschema.NewStructural(&apiextensions.JSONSchemaProps{
```

Comment:
check error return here and fail the test if we get an error (same for the next call to NewStructural)

Reply:
or, it might be more readable / accurate to declare the schema in a yaml string and unmarshal it to a CRD, like this:

https://github.com/kubernetes/kubernetes/blob/2b203348fbe7d05dec9e913e0fb50045a2474425/staging/src/k8s.io/apiextensions-apiserver/test/integration/listtype_test.go#L59-L74

https://github.com/kubernetes/kubernetes/blob/2b203348fbe7d05dec9e913e0fb50045a2474425/staging/src/k8s.io/apiextensions-apiserver/test/integration/listtype_test.go#L130-L132


and explicitly declare the `numArray` property as a list rather than using AdditionalProperties

Reply:
```
const listTypeResourceSchema = `
type: object
properties:
  numArray:
    type: array
    XListType: set
    XMapType: atomic
`
```
`XListType` works for me.

```
x-kubernetes-list-type: map 
```
`x-kubernetes-list-type` cannot be unmarshal to the object correctly.